### ISSUE TITLE
CppMNIST

### ISSUE THREAD

**Issue by euxsun** [emstoudenmire/TNML#1]
Hello!
Trying to run the code I've reached CppMNIST absence problem. How it should be organized? Where I can take the folder sources?

Sincerely,
Evgeniy Pavlovskiy, leading researcher @ SDAML NSU lab.

**Comment by emstoudenmire** [comment]
Hi Evgeniy,
Sorry for the very slow reply. Which code were you trying to compile? I just compiled all of the codes and did not get any compilation errors. Did you clone/pull the latest version of this repo? 

Also, if I may add, these codes were intended as research codes i.e. to do proof-of-principle calculations and do not represent the most efficient way of optimizing tensor networks for machine learning. A much much efficient way is to use stochastic gradient descent with mini batches. So if you are really interested in running experiments, may I suggest the following very new library that builds on top of PyTorch: https://github.com/jemisjoky/TorchMPS

Best regards,
Miles

**Comment by euxsun** [comment]
Sure, it complies good. But there is a line in the code:
`auto datadir = input.getString("datadir","/Users/mstoudenmire/software/tnml/CppMNIST");`
Where I can take CppMNIST structure? How it differs from usual MNIST? Or may be it is in mllib directory?

**Comment by emstoudenmire** [comment]
Hi, so the value of "datadir" should be the folder where the MNIST data files are located. That happened to be where I had it on my computer (so you can ignore the fact that the folder was called CppMNIST). I'll update the codes to put a more generic default for datadir - thanks for pointing this out.
